exotic/tests/test_utils.py

```diff
@@ -0,0 +1,58 @@
+from exotic.utils import round_to_2
+
+
+class TestRoundToTwo:
+    """tests the round_to_2() function"""
+
+    _ARBITRARY_NUMBER = 4.0
+
+    def test_with_one_arg(self):
+        # One arg may have been passed in
+        result = round_to_2(self._ARBITRARY_NUMBER)
+        assert self._ARBITRARY_NUMBER == result
+
+    def test_second_arg_special_zero_case(self):
+
+        # when second argument passed in is 0.0.
+        result = round_to_2(self._ARBITRARY_NUMBER, 0.0)
+        assert self._ARBITRARY_NUMBER == result
+
+    def test_round_to_two_decimal_places(self):
+
+        float_with_long_fractional_part = 2.34567
+        result = round_to_2(float_with_long_fractional_part, 2)
+        assert 2.35 == result
+
+        result = round_to_2(2.000123456)
+        assert 2.0 == result
+
+        # NOTE: it's kind of weird that the second argument would
+        # just be discarded here. Fix it later once I understand
+        # the code better
+        result = round_to_2(float_with_long_fractional_part, 3)
+        assert 2.35 == result
+
+    def test_small_numbers(self):
+        # the meat of the function. This gets into testing the
+        # -int(floor(log10(abs(y)))) expression
+
+        # for numbers where log10(n) * -1 is 1, round to two places
+        result = round_to_2(0.1234567890)
+        assert 0.12 == result
+
+        # for numbers that are very small, round to two sig figs
+        result = round_to_2(0.000123456789)
+        assert 0.00012 == result
+
+        # for numbers that are very small, and negative, round to two sig figs and keep the negativity
+        result = round_to_2(-0.000123456789)
+        assert -0.00012 == result
+
+        # not sure if this is a bug?
```

Comment:
doesn't look like a bug to me- it takes the second number, looks for 2 sig figs (non-0s after the decimal), so in this case that would be 0.00012, and then rounds the first number to that number of sig figs - which should be 2.12300 - but the trailing 0s get dropped by Python

Reply:
good to know. I wasn't sure if more precision would be needed. I'll update to remove this comment.